LoopKit/UserAlert.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  UserAlert.swift
+//  LoopKit
+//
+//  Created by Rick Pasetto on 4/8/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+/// Protocol that describes any class that handles Alert posting.
+public protocol UserAlertHandler: class {
+    /// Schedule the given alert for posting.
+    func scheduleAlert(_ alert: UserAlert)
+    /// Unschedule any pending alerts with the given identifier.
+    func unscheduleAlert(managerIdentifier: String, typeIdentifier: UserAlert.TypeIdentifier)
+    /// Remove any alerts currently posted with the given identifier.  It ignores any pending alerts.
+    func cancelAlert(managerIdentifier: String, typeIdentifier: UserAlert.TypeIdentifier)
+}
+
+/// Protocol that describes something that can deal with a user's response to an alert.
+public protocol UserAlertResponder {
+    /// Acknowledge alerts with a given type identifier
+    func acknowledgeAlert(typeIdentifier: UserAlert.TypeIdentifier) -> Void
+}
+
+/// Structure that represents an Alert that needs to be shown to the User.
+public struct UserAlert {
+    /// Representation of an alert Trigger
+    public enum Trigger {
+        /// Trigger the alert immediately
+        case immediate
+        /// Delay triggering the alert by `interval`, but issue it only once.
+        case delayed(interval: TimeInterval)
```

Comment:
Do we have use cases from devices for `delayed` and `repeating`?

Reply:
@ps2 Yes.  In Dexcom, SignalLossAlert is `repeating` and TransmitterNotFound is `delayed.`